chain/chain/src/state_snapshot_actor.rs

```diff
@@ -90,7 +100,20 @@ impl actix::Handler<WithSpanContext<CreateSnapshotRequest>> for StateSnapshotAct
             tracing::error!(target: "state_snapshot", ?prev_block_hash, ?shard_uids, "Failed to unlock flat state updates");
         }
         match res {
-            Ok(_) => {
+            Ok(res_shard_uids) => {
+                if let Some(res_shard_uids) = res_shard_uids {
+                    self.network_adapter.send(PeerManagerMessageRequest::NetworkRequests(
+                        NetworkRequests::SnapshotHostInfo {
```

Comment:
Reviewer(s): please double check the properties of the StateSnapshot recorded here.

Reply:
Took me more than a moment to realize what's happening to the set of shard_uids. Looks good.